FIAT/dual_set.py

```diff
@@ -54,13 +57,67 @@ def to_riesz(self, poly_set):
         tshape = self.nodes[0].target_shape
         num_nodes = len(self.nodes)
         es = poly_set.get_expansion_set()
+        ed = poly_set.get_embedded_degree()
         num_exp = es.get_num_members(poly_set.get_embedded_degree())
+        nbf = poly_set.get_num_members()
 
         riesz_shape = tuple([num_nodes] + list(tshape) + [num_exp])
 
         self.mat = numpy.zeros(riesz_shape, "d")
 
-        for i in range(len(self.nodes)):
-            self.mat[i][:] = self.nodes[i].to_riesz(poly_set)
+        # let's amalgamate the pt_dict and deriv_dicts of all the
```

Comment:
Maybe I'm just to dim for this, but I find this code unintelligible. I think a key problem is that the local variable names are so abbreviated that I can't understand what is what.

Reply:
I've added some top-level documentation explaining a bit about what it does.